develop/tutorials/articles/03-developing-a-web-application/04-writing-web-application-liferay/01-writing-a-web-application.markdown

```diff
@@ -0,0 +1,635 @@
+
+This command creates a project named `com.liferay.docs.guestbook`. `-c
+GuestbookPortlet` specifies the name of the class to be generated. `-p
+com.liferay.docs.guestbook` specifies the package in which the class is to be
+generated. By default, the `mvcportlet` template is used. You can explicitly
+specify this using the `-t mvcportlet` option. When you use the `mvcportlet`
+template, Blade Tools creates a portlet class (the name of which can be
+specified with the `-c` option) in a package (the name of which can be
+specified by the `-p` option) in your project's `src/main/java` folder. Blade
+Tools also creates your application's JSP files and language properties file in
+your project's `src/main/resources` folder.
+
+After you've created your project, import it into Liferay IDE.
+
+Note: Two important concepts for Liferay 7 development are modules and
+components. A *module* is the one and only type of Liferay 7 plugin. A
+*component* is an implementation of an interface. In Liferay, a component is
+typically a
+[Declarative Services](http://wiki.osgi.org/wiki/Declarative_Services)
+component, which means that it's a Java class that's decorated with the
+`org.osgi.service.component.annotations.Component` annotation. Please see the [Understanding Liferay's Module Framework](/develop/tutorials/-/knowledge_base/7-0/understanding-liferays-module-framework)
+tutorial for more information on Liferay modules, components, and Liferay's
+module framework. In this learning path, you're creating a single module called
+`com.liferay.docs.guestbook`. In this module, you're creating a single
+component: the `GuestbookPortlet` class which implements the `Portlet`
+interface.
+
+## What is a Portlet?
+
+When you access a web site, you interact with an application. That application
+might be simple: it might only show you one piece of information, such as an
+article. The application might be complex: you could be doing your taxes
+online, inputting lots of data into an application that calculates whether you
+owe or are due a refund. These applications run on a *platform* that provides
+application developers the building blocks they need to make applications.
+
+![Figure x: Many Liferay applications can run at the same time on the same page.](../../images/portlet-applications.png)
+
+Liferay Portal is a platform, and its applications are called *portlets*. One
+difference between a Liferay application and an application built on another
+platform is that Liferay Portal can serve many applications on the same page at
+the same time. Usually, a web application takes up the entire page. If you
+want, you can do this with Liferay as well. But Liferay has the added benefit
+of being able to run many applications on the same page. Liferay's framework
+takes this into account at every step. For example, features like
+platform-generated URLs exist to support Liferay's ability to serve multiple
+applications on the same page.
+
```

Comment:
We want to be careful with this. In the new world, we present portlets as one possible front-end for an application. The developer could just as easily generate services and business logic, and then write a screens app that interfaces with the server.